toplevel/vernacentries.ml

```diff
@@ -102,17 +102,19 @@ let try_print_subgoals () =
 let show_intro all =
   let pf = get_pftreestate() in
   let {Evd.it=gls ; sigma=sigma; } = Proof.V82.subgoals pf in
-  let gl = {Evd.it=List.hd gls ; sigma = sigma; } in
-  let l,_= decompose_prod_assum (strip_outer_cast (pf_concl gl)) in
-  if all
-  then
-    let lid = Tactics.find_intro_names l gl in
-    msg_notice (hov 0 (prlist_with_sep  spc pr_id lid))
-  else
-    try
-      let n = List.last l in
-      msg_notice (pr_id (List.hd (Tactics.find_intro_names [n] gl)))
-    with Failure "List.last" -> ()
+  if gls <> [] then begin
+    let gl = {Evd.it=List.hd gls ; sigma = sigma; } in
+    let l,_= decompose_prod_assum (strip_outer_cast (pf_concl gl)) in
+    if all
+    then
+      let lid = Tactics.find_intro_names l gl in
+      msg_notice (hov 0 (prlist_with_sep  spc pr_id lid))
+    else
+      try
+        let n = List.last l in
+        msg_notice (pr_id (List.hd (Tactics.find_intro_names [n] gl)))
+      with Failure "List.last" -> ()
```

Comment:
Is this the occasion to rewrite this dubious pattern of catching a string exception? `else if l <> []` should do the trick.